### PR TITLE
Add doc Neuron models

### PR DESCRIPTION
**Pull request by JingyaHuang**
- [x] Neuron Model API doc
- [x] Neuron Model usage guide
- [x] Update model exporting guide
- [x] Patch snippets in modeling docstring

### CODE REVIEW

File: docs/source/guides/export_model.mdx [hunk 1]
```diff
@@ -168,42 +170,35 @@ optimum-cli export neuron --model local_path --task question-answering --batch_s
...
-logits = neuron_model(*tuple(example.values()))
+>>> from optimum.neuron import NeuronModelForSequenceClassification
+
+>>> input_shapes = {"batch_size": 1, "sequence_length": 64}  # mandatory shapes
+>>> model = NeuronModelForSequenceClassification.from_pretrained(
+...   "distilbert-base-uncased-finetuned-sst-2-english", export=True, **input_shapes
+... ) 
+
+# Save the model
+>>> model.save_pretrained("./distilbert-base-uncased-finetuned-sst-2-english_neuron/")
```
Comment (philschmid) [inline docs/source/guides/export_model.mdx:186]: does that save the config?

File: docs/source/guides/models.mdx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# Neuron Model Inference
+
+*The APIs presented in the following documentation are relevant for the inference on [INF2](https://aws.amazon.com/ec2/instance-types/inf2/),
```
Comment (philschmid) [inline docs/source/guides/models.mdx:19]: Its Inf2, Inf1, Trn1

File: docs/source/guides/models.mdx [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+-->
+
+# Neuron Model Inference
+
+*The APIs presented in the following documentation are relevant for the inference on [INF2](https://aws.amazon.com/ec2/instance-types/inf2/), 
+[TRN1](https://aws.amazon.com/ec2/instance-types/trn1/) and [INF1](https://aws.amazon.com/ec2/instance-types/inf1/).*
+
+[🤗 Optimum-neuron](https://github.com/huggingface/optimum-neuron) is a utility package to ease the use of accelerators like AWS Inferentia2, 
+Trainium1 and Inferentia1. With Neuron models APIs, you will be able to run inference on neuron devices with the same experience as 
+[🤗 Transformers](https://huggingface.co/docs/transformers/index).
```
Comment (philschmid) [inline docs/source/guides/models.mdx:24]:
```suggestion
```

File: docs/source/guides/models.mdx [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+```diff
+from transformers import AutoTokenizer
+-from transformers import AutoModelForSequenceClassification
++from optimum.neuron import NeuronModelForSequenceClassification
+
+# PyTorch checkpoint
+-model = AutoModelForSequenceClassification.from_pretrained("distilbert-base-uncased-finetuned-sst-2-english") 
+
+# Compile your model during the first time
++input_shapes = {"batch_size": 1, "sequence_length": 64}
```
Comment (philschmid) [inline docs/source/guides/models.mdx:45]: is batch_size default 1? 
could we also do? 
```
model = NeuronModelForSequenceClassification.from_pretrained(
    "distilbert-base-uncased-finetuned-sst-2-english", export=True, sequence_length=123
) 
```?

File: optimum/neuron/modeling.py [hunk 5]
```diff
@@ -86,12 +87,12 @@
     >>> tokenizer = {processor_class}.from_pretrained("{checkpoint}")
     >>> model = {model_class}.from_pretrained("{checkpoint}")
 
-    >>> inputs = tokenizer("My name is Philipp and I live in Germany.", return_tensors="pt")
```
Comment (philschmid) [inline optimum/neuron/modeling.py:89]: :(

**Review by philschmid — APPROVED** [review]
LGTM. 
I left two minor suggestions

File: docs/source/guides/export_model.mdx (hunk 1 above)
Comment (JingyaHuang) [reply docs/source/guides/export_model.mdx:186]: Yes it does: https://github.com/huggingface/optimum/blob/main/optimum/modeling_base.py#L124

**Pull request merged**